### ISSUE TITLE
Multiple ConfigurationKey for Consul Store KV

### ISSUE THREAD

**Issue by Stronhold** [ThreeMammals/Ocelot#384]
I would like to know if there can bemultiple configuration keys to access different configuration files in KV Store or if I can match them using a pattern like it is done in [issue 296](https://github.com/ThreeMammals/Ocelot/issues/296)

**Comment by TomPallister** [comment]
Hi @Stronhold please see docs [here](http://ocelot.readthedocs.io/en/latest/features/configuration.html#configuration-key) this might help!

**Comment by Stronhold** [comment]
Hi,

I checked that before. As i Saw, there Gould be only one configurationkey at the same time or am I wrong?

Thanks un advance

**Comment by TomPallister** [comment]
@Stronhold you can have as many keys as you want per deployment of Ocelot. You cannot merge multiple configurations from Consul at the moment. If you would like to implement this I can support you! 😄

**Comment by Stronhold** [comment]
That was the point! Any help would be appreciated. 
Morover, i would like yo know if when we use the administration API, the config.json file, could be overwritten? As far as I have tested, this does not happen 

Thanks in advance!

**Comment by TomPallister** [comment]
@Stronhold are you using admin api with consul storing your configuration? If so the json file is not changed. If you are not using consul then it will be changed.

**Comment by Stronhold** [comment]
Nope. The idea would be using multiple configurationkey and use consul to have the routes. Through the administration API chance  the config.json. 

Any support to have multiple configurationkey is appreciated :)

**Comment by TomPallister** [comment]
Sorry @Stronhold I am very confused by what it is you want to implement. Please could you add as much detail as possible so I can understand.

**Comment by Stronhold** [comment]
I want to be able to have múltiple configurationkey to access to different consul configurations and merge them

**Comment by TomPallister** [comment]
this feature implement config merging for consul, same logic we use for file merging but applied to consul.

**Comment by marrop** [comment]
This might be slightly off topic but here goes. Is it possible via Ocelot to read a consul value from Consul's Key/Value store? Or is the only value one may read from Consul the configuration ("InternalConfiguration" or set by "GlobalConfiguration/ConfigurationKey"?

**Comment by raman-m** [comment]
@ggnaegi  What is your opinion on this crazy idea with the Consul config? 😄

**Comment by ggnaegi** [comment]
> @ggnaegi  What is your opinion on this crazy idea with the Consul config? 😄 

I think it would be a nice addition, at least provide a custom key for the ocelot config stored in key/value store.

But then merge config... Like appsettings?

**Comment by raman-m** [comment]
@Stronhold  Dear Sergio!
Your request confuses all parties including Tom.
What is your user scenario?

Also, I see that [your repos](https://github.com/Stronhold?tab=repositories) have no Ocelot forked repo.
So, forked [Ocelot repo](https://github.com/Stronhold/Ocelot) doesn't exist in your account!
I would say you  violated Ocelot [development process](https://ocelot.readthedocs.io/en/latest/building/releaseprocess.html) which says: discuss -> issue -> PR.
But working on an issue means you have to fork Ocelot repo.
🆗 First fork, then request...
You have a right to reopen the issue in future.
